### PR TITLE
Bug/bill cpc 746 change display id to names

### PR DESCRIPTION
**Pull request by ebond07**
JIRA: link to jira ticket
## Context:
The main page for bills (bill history) displayed a user Id and vet Id which meant nothing (they aren't the ones used as parameters in requests). So I switched them to instead display the owner's and vet's full names. Also had to change the type of these two attributes to match the type of their corresponding attribute in their respective microservice. This was done to facilitate later changes involving the two attributes.
## Changes
- Owner Name column added
- Owner Name column now displays owner's full name 
- Vet ID column added
- Vet Name column now displays vet's full name 
- CustomerId is now a string instead of an integer throughout the billling service and api-gateway
- VetId is like customerId now a string instead of an integer.
- Added a method to get the vet entity through matching the vetBillId attribute of the vet class and the vetId attribute of the bill class
- Added a method to get the customer's name through mapping the names with an id and then comparing that id to the customerId bill holds
## Before and After UI (Required for UI-impacting PRs)
Before
Screenshot 2023-09-24 201841
After
Screenshot 2023-09-25 193929

### CODE REVIEW

File: api-gateway/src/main/java/com/petclinic/bffapigateway/dtos/Bills/BillDetails.java [hunk 1]
```diff
@@ -1,22 +1,16 @@
...
-
-import static java.util.stream.Collectors.toList;
 
 @Builder
 @Data
 public class BillDetails {
     private String billId;
     private LocalDate date;
-    private int customerId;
+    private String customerId;
```
Comment (drallo22) [inline api-gateway/src/main/java/com/petclinic/bffapigateway/dtos/Bills/BillDetails.java:13]: Good job changing it to a string makes so much more sense.

File: api-gateway/src/main/resources/static/scripts/bill-history/bill-history.controller.js [hunk 2]
```diff
@@ -6,6 +6,26 @@ angular.module('billHistory')
         let self = this;
         self.billHistory = []
 
+        self.owners = [
```
Comment (drallo22) [inline api-gateway/src/main/resources/static/scripts/bill-history/bill-history.controller.js:9]: If hardcoding was the only way to solve this problem, maybe we should consider using an aggregator

**Review by drallo22 — APPROVED** [review]
Looks good! As I said in the comments maybe we should consider using an aggregator, but looks good for what you were able to do considering the circumstances!

**Review by dahlica12 — APPROVED** [review]
Very useful feature to have, UI is more user-friendly and readable

**Review by HennaCH — APPROVED** [review]
The front end is more user friendly now, good job!

**Pull request merged**